api/internal/git/repospec_test.go

```diff
@@ -443,18 +443,6 @@ func TestNewRepoSpecFromUrl_Smoke(t *testing.T) {
 				GitSuffix: ".git",
 			},
 		},
-		{
-			name:      "non_parsable_path",
-			input:     "https://authority/org/repo/%-invalid-uri-so-not-parsable-by-net/url.Parse",
-			cloneSpec: "https://authority/org/repo.git",
-			absPath:   notCloned.Join("%-invalid-uri-so-not-parsable-by-net/url.Parse"),
-			repoSpec: RepoSpec{
-				Host:      "https://authority/",
-				OrgRepo:   "org/repo",
-				Path:      "%-invalid-uri-so-not-parsable-by-net/url.Parse",
-				GitSuffix: ".git",
-			},
-		},
```

Comment:
Why did you remove this test? Even if we might change it in the future, it is good to document current behaviour and make the change obvious on the future PR.

Reply:
I can add it back. 

My (I guess false) assumption was to add a test only if the behavior is intentional. It's why I question my code changes when they break existing tests. I removed this test because we weren't decided on the behavior.

Reply:
I'd say that while we haven't decided on whether/how to change the behaviour, there is a current behaviour that exists in Kustomize today, and having this test will just make it explicit that we're changing it. We actually have a [doc](https://kubectl.docs.kubernetes.io/contributing/kustomize/bugs/#if-you-have-go-installed) encouraging folks reporting bugs to submit passing tests that demonstrate them, albeit with inline comments on how/why the behaviour the test shows is wrong.

Reply:
Got it. Thanks for the explanation!